internal/service/fw/api_internal_domain_lists_resource.go

```diff
@@ -0,0 +1,168 @@
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "",
+		Attributes:          AtcfwInternalDomainsResourceSchemaAttributes,
+	}
+}
+
+func (r *InternalDomainListsResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*bloxoneclient.APIClient)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *bloxoneclient.APIClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *InternalDomainListsResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data AtcfwInternalDomainsModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiRes, _, err := r.client.FWAPI.
+		InternalDomainListsAPI.
+		InternalDomainListsCreateInternalDomains(ctx).
+		Body(*data.Expand(ctx, &resp.Diagnostics)).
+		Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create InternalDomainLists, got error: %s", err))
+		return
+	}
+
+	res := apiRes.GetResults()
+	data.Flatten(ctx, &res, &resp.Diagnostics)
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *InternalDomainListsResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data AtcfwInternalDomainsModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiRes, httpRes, err := r.client.FWAPI.
+		InternalDomainListsAPI.
+		InternalDomainListsReadInternalDomains(ctx, int32(data.Id.ValueInt64())).
+		Execute()
+	if err != nil {
+		if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read InternalDomainLists, got error: %s", err))
+		return
+	}
+
+	res := apiRes.GetResults()
+	data.Flatten(ctx, &res, &resp.Diagnostics)
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *InternalDomainListsResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data AtcfwInternalDomainsModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiRes, _, err := r.client.FWAPI.
+		InternalDomainListsAPI.
+		InternalDomainListsUpdateInternalDomains(ctx, int32(data.Id.ValueInt64())).
```

Comment:
The name here is so redundant. We should change this in the client code. Not for this PR though.

Reply:
This pattern follows for all the Objects , once we are through with DFP , we can change this for all objects